src/utils/utils.ts

```diff
@@ -426,3 +427,37 @@ export function wrapInASCIIBox(text: string, padding = 0): string {
 export function isFilePayload(value: any): boolean {
   return typeof value === 'object' && value['name'] && value['mimeType'] && value['buffer'];
 }
+
+
+function toMegabytes(bytes: number) {
+  const mb = bytes / 1024 / 1024;
+  return `${Math.round(mb * 10) / 10} Mb`;
+}
+
+export function getDownloadProgress(progressBarName: string) {
+
+  let progressBar: ProgressBar;
+  let lastDownloadedBytes = 0;
+
+  return (downloadedBytes: number, totalBytes: number) => {
+    if (!process.stderr.isTTY) return;
+    if (!progressBar) {
+      progressBar = new ProgressBar(
+          `Downloading ${progressBarName} - ${toMegabytes(
+              totalBytes
+          )} [:bar] :percent :etas `,
+          {
+            complete: '=',
+            incomplete: ' ',
+            width: 20,
+            total: totalBytes,
+          }
+      );
+    }
+    const delta = downloadedBytes - lastDownloadedBytes;
+    lastDownloadedBytes = downloadedBytes;
+    progressBar.tick(delta);
+  };
+}
+
+export const getErrorMessage = (error: Error) => typeof error === 'object' && typeof error.message === 'string' ? error.message : '';
```

Comment:
Let's instead inline this logic into `downloadFile` and return `{ errorMessage }`.

Reply:
https://github.com/microsoft/playwright/blob/0908dc98c8d6226fa02c6a0a5fd81003dcb8e158/src/server/browserType.ts#L84

The functionality is duplicated here, should we keep this in utils and use it in BrowserType too?

Reply:
I think it's better to be focused in this PR 😄